src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/UserRoleServiceTests.cs

```diff
@@ -70,23 +77,139 @@ await using (var userAndRolesDbContext = InMemoryUserAndRolesDbContext(userAndRo
+            {
+                await userAndRolesDbContext.AddAsync(user);
+                await userAndRolesDbContext.AddAsync(role);
+                await userAndRolesDbContext.SaveChangesAsync();
+            }
+
+            var userManager = MockUserManager();
+
+            userManager
+                .Setup(mock => mock.GetRolesAsync(
+                    ItIsUser(user)))
+                .ReturnsAsync(ListOf(role.Name));
+            
+            await using (var userAndRolesDbContext = InMemoryUserAndRolesDbContext(userAndRolesDbContextId))
+            {
+                var service = SetupUserRoleService(
+                    usersAndRolesDbContext: userAndRolesDbContext,
+                    userManager: userManager.Object);
+
+                var result = await service.SetGlobalRole(user.Id, role.Id);
+
+                VerifyAllMocks(userManager);
+
+                result.AssertRight();
+
+                VerifyAllMocks(userManager);
+            }
+        }
+
+        [Fact]
+        public async Task SetGlobalRole_AlreadyHasAnotherRole()
+        {
+            var user = new ApplicationUser
+            {
+                Id = Guid.NewGuid().ToString()
+            };
+
+            var newRole = new IdentityRole
+            {
+                Id = Guid.NewGuid().ToString(),
+                Name = "New Role"
+            };
+            
+            var existingRole = new IdentityRole
+            {
+                Id = Guid.NewGuid().ToString(),
+                Name = "Existing Role"
+            };
+
+            var userAndRolesDbContextId = Guid.NewGuid().ToString();
+
+            await using (var userAndRolesDbContext = InMemoryUserAndRolesDbContext(userAndRolesDbContextId))
+            {
+                await userAndRolesDbContext.AddAsync(user);
+                await userAndRolesDbContext.AddRangeAsync(newRole, existingRole);
+                await userAndRolesDbContext.SaveChangesAsync();
+            }
+
+            var userManager = MockUserManager();
+
+            // Here we are setting up the user so that they have a different Global Role currently assigned
+            // than the new one being set.  They will have the existing one removed and the new one added.
+            userManager
+                .Setup(mock => mock.GetRolesAsync(
+                    ItIsUser(user)))
+                .ReturnsAsync(ListOf(existingRole.Name));
+
+            userManager
+                .Setup(mock => mock.AddToRoleAsync(
+                    ItIsUser(user), newRole.Name))
+                .ReturnsAsync(new IdentityResult());
+            
+            // TODO DW - do we need to handle this?
+            userManager
+                .Setup(mock => mock.RemoveFromRolesAsync(
+                    ItIsUser(user), ItIs.ListSequenceEqualTo(ListOf(existingRole.Name))))
+                .ReturnsAsync(new IdentityResult());
```

Comment:
I think you do need this, but would it also help for the future to add `MockBehavior.Strict` to `AdminMockUtils.MockUserManager` when setting it up? Just ran all tests with it and it seems like that's an easy change to make right now with no failing tests.

Reply:
Sure I tried this before and got some failures, but I have added it in and it seems all good, so good suggestion!